scripts/irods/test/test_iadmin.py

```diff
@@ -1685,6 +1685,43 @@ def test_iadmin_mkgroup_returns_correct_error_on_duplicate_group__issue_7599(sel
         finally:
             self.admin.run_icommand(['iadmin', 'rmgroup', test_group_name])
 
+    def test_iadmin_exits_nonzero_on_error_cases__issue_7734(self):
+        # Nonexistent command
+        out, err, ec = self.admin.run_icommand(['iadmin', 'nonexistentcommand']);
+        self.assertNotEqual(ec, 0)
+
+        # Bad parameters to modrepl
+        out, err, ec = self.admin.run_icommand(['iadmin', 'modrepl', 'foo', 'bar', 'baz']);
+        self.assertNotEqual(ec, 0)
+
+        # Nonexistent data object 
+        out, err, ec = self.admin.run_icommand(['iadmin', 'modrepl', 'logical_path', '/tempZone/object/does/not/exist', 'replica_number', '0', 'DATA_COMMENTS', 'something']);
+        self.assertNotEqual(ec, 0)
+
+        try:
+            test_file_name = 'test_file_name_issue_7734'
+            self.admin.assert_icommand(['itouch', self.admin.home_collection + '/' + test_file_name])
+
+            # Nonexistent replica number
+            out, err, ec = self.admin.run_icommand(['iadmin', 'modrepl', 'logical_path', self.admin.home_collection + '/' + test_file_name, 'replica_number', '123', 'DATA_COMMENTS', 'something']);
+            self.assertNotEqual(ec, 0)
+
+            # Invalid replica number
+            out, err, ec = self.admin.run_icommand(['iadmin', 'modrepl', 'logical_path', self.admin.home_collection + '/' + test_file_name, 'replica_number', 'abc', 'DATA_COMMENTS', 'something']);
+            self.assertNotEqual(ec, 0)
+
+        finally:
+            self.admin.run_icommand(['irm', '-f', self.admin.home_collection + '/' + test_file_name])
```

Comment:
I believe this is covered in part by `test_iadmin.Test_Iadmin_modrepl.test_input_parameters_invalid` and/or these assertions would fit nicely into that test. Perhaps we can move this test into that class or modify the existing test to include assertions around the return code since this seems mainly to deal with `iadmin modrepl` (except for the first assertion, which should perhaps stay in a test on its own). Thoughts?

Reply:
Yeah, moving this test into the class sounds good to me. I like the idea of keeping these assertions together since they target a specific issue.